### PR TITLE
Enabling GuestAgent Logging

### PR DESCRIPTION
**Pull request by arisettisanjana**
Currently only backup initiated extension logs (of only last successful retry of one run) are sent to Kusto -TraceLogMessages. RestorePoint initiated extension job's logs are not available in Kusto. To obtain logs of each extension run, we utilize the event logging feature provided by Guest Agent, push extension telemetry to AzCore Kusto cluster's fa database. 
So, this feature was previously disabled as we have seen a regression, reactivating the logging functionality through the guest agent by removing the asynchronous programming logic (which still can be triggered based on the config flag) that triggered the event every 10 seconds to capture the logs.

### CODE REVIEW

**Comment by iamashwani-who** [comment]
Can you please update the description to add more details about what is the theory and why are trying this out?

File: VMBackup/main/handle.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def main():
         hutil.patching = MyPatching
         configSeqNo = -1
         hutil.try_parse_context(configSeqNo)
-        disable_event_logging = hutil.get_intvalue_from_configfile("disable_logging", 1)
+        disable_event_logging = hutil.get_intvalue_from_configfile("disable_logging", 0)
```
Comment (iamashwani-who) [inline VMBackup/main/handle.py:82]: just wondering if it is possible to add another setting to allow for in-process (without another thread) or out-of-process (with a separate thread) processing of logs. That could allow you to switch and perform more validations as necessary.

**Comment by deveshjagwani** [comment]
> Can you please update the description to add more details about what is the theory and why are trying this out?

+1 to Ashwani's comment. Please add detailed description, its effects, testing, etc.

**Comment by deveshjagwani** [comment]
Is it possible to test this change somehow end-to-end internally ? I am looking for some form of testing.

**Comment by arisettisanjana** [comment]
yes it is possible to test this changes end-to-end, have done the testing and see the logs in Kusto

File: VMBackup/main/Utils/EventLoggerUtil.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ def __init__(self, event_directory, severity_level):
         self.event_logging_error_count = 0
         self.events_folder = event_directory
         self.event_logging_enabled = bool(self.events_folder)
+        self.use_threads = use_async_event_logging
```
Comment (iamashwani-who) [inline VMBackup/main/Utils/EventLoggerUtil.py:35]: let us call this variable async_event_logging locally too instead of use_threads?

**Comment by deveshjagwani** [comment]
> yes it is possible to test this changes end-to-end, have done the testing and see the logs in Kusto

**Review by iamashwani-who — APPROVED** [review]

**Review by deveshjagwani — APPROVED** [review]

**Pull request merged**